pkg/liqo-controller-manager/networking/internal-network/client-controller/client_controller.go

```diff
@@ -106,9 +106,14 @@ func (r *ClientReconciler) ensureInternalFabric(ctx context.Context, gwClient *n
 		return fmt.Errorf("internal endpoint not found for the gateway client %q", gwClient.Name)
 	}
 
+	internalFabricName, err := getters.RetrieveClusterIDFromObject(gwClient)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve the cluster ID from the gateway client %q", gwClient.Name)
+	}
```

Comment:
In that way, if we update the liqo controllers, we will have two internal fabrics created for each remote cluster. Can we check that no other internal fabric exists for the same cluster ID? In that way, we can support a non-breaking update for this version. The same for the other controller